src/languages/java.md

```diff
@@ -1,3 +1,101 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is a general-purpose programming language, and one of the most popular in the world today. Platform.sh supports Java runtimes that can be used with build management tools such as Gradle, Maven, and Ant.
+
+
+## Supported versions
+
+* 8
+* 11
+* 12
+
+
+To select a Java version, specify a `type` such as `java:11`:
+
+```yaml
+# .platform.app.yaml
+type: "java:11"
+```
+
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) which handles decoding of service credential information for you.
+
+## Support build automation
+
+Platform.sh has support for software project management and comprehension tool in the Java World: 
+
+* [Gradle](https://gradle.org/)
+* [Maven](https://maven.apache.org/)
+* [Ant](https://ant.apache.org/)
+
+
+
+## Accessing services
+
+To access various [services](/configuration/services.md) with Java, see the following examples.  The individual service pages have more information on configuring each service.
+
+
+
+{% language name="MongoDB", type="php", url="https://examples.docs.platform.sh/php/mongodb" -%}
+
+{% language name="MySQL", type="php", url="https://examples.docs.platform.sh/php/mysql" -%}
+
+{% language name="PostgreSQL", type="php", url="https://examples.docs.platform.sh/php/postgresql" -%}
+
+{% language name="Redis", type="php", url="https://examples.docs.platform.sh/php/redis" -%}
+
+<!--
+
+{% language name="RabbitMQ", type="php", url="https://examples.docs.platform.sh/php/rabbitmq" -%}
+
+{% language name="Solr", type="php", url="https://examples.docs.platform.sh/php/solr" -%}
+
+{% codetabs name="Elasticsearch", type="php", url="https://examples.docs.platform.sh/php/elasticsearch" -%}
+
+{% language name="Memcached", type="php", url="https://examples.docs.platform.sh/php/memcached" -%}
+-->
```

Comment:
Commented block can be removed.

Reply:
I want to keep this on someplace to don't forget.
Do we have Jira? Can we use Github Project?